docs/configuring-archival.md

```diff
@@ -0,0 +1,135 @@
+2. **Debugging:** Old Event Histories may help with debugging.
+
+:::info
+
+**Want to turn off the Archival feature?**
+
+Set `archival.history.state` and `namespaceDefaults.archival.history.state` to `disabled`.
+
+:::
+
+## How to use Archival
+
+There are four main components to the Archival feature that you need to be aware of.
+
+1. **Archival feature configuration**: The Archival feature is controlled by the `archival` configuration in the `config/development.yaml` file.
+2. **Archive Provider**: The location where the Event Histories are archived to is called the Archive Provider. Common providers are S3, GCloud, Kafka, and the local file system.
+3. **Archiver**: The actual system component that archives and retrieves Event Histories is called an Archiver. An Archiver is directly correlated to a single Archive Provider. 
+4. **URI**: The URI is what the Namespace uses to know which Archive Provider and Archiver to use. The schema of the URI identifies which Archiver and Provider to use, and where the URI points to, identifies the actual archive location.
+
+Temporal supports several Archive Providers (and Archivers) out of the box as well as the option to create and use your own. If you want to create your own Archiver, follow [this guide](https://github.com/temporalio/temporal/blob/master/common/archiver/README.md).
+
+New Namespaces require the Archival URI to be set at the time of creation. Once set, the URI can not be changed for a Namespace. Each Namespace supports a single Archival URI, but each Namespace in a cluster can have a unique URI.
+
+When creating a new Namespace, if the URI is not specifically provided, the Namespace will use the value of `defaultNamespace.archival.history.URI` from the `config/development.yaml` file.
+
+A Namespace can safely switch between `enabled` and `disabled` states, however `archival.history.state` must be set to `"enabled"` for Archival to work within any Namespace within the cluster regardless of whether the Namespace has Archival enabled.
+
+Currently, the `defaultNamespace` configuration makes use of the local file system Provider and Archiver, storing Event Histories at `file:///tmp/temporal_archival/development`.
+
+However, in a production environment, you will most likely want to store Event Histories in a cloud based storage system such as S3 or Google Cloud. In this scenario you will need to make changes in the `config/development.yaml` file or use the dynamic config overwrite.
+
+:::note
+
+The dynamic config will only take effect if `archival.history.state` is set to `"enabled"` in `config/development.yaml`.
+
+:::
+
+Archival is supported in global Namespaces (Namespaces spanning multiple clusters). When Archival runs for a global Namepsace, it will first run on the active cluster and then some time later it will run on the standby cluster as well. Before archiving, a history check is done to see what has already been uploaded.
```

Comment:
```suggestion
Archival is supported in global Namespaces (Namespaces spanning multiple clusters). When Archival is running in a global Namepsace, it will first run on the active cluster and then some time later it will run on the standby cluster. Before archiving, a history check is done to see what has been previously been uploaded.
```